AssetHelper/CatalogTools/CatalogEntry.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using UnityEngine.AddressableAssets.ResourceLocators;
+using UnityEngine.ResourceManagement.ResourceLocations;
+using UnityEngine.ResourceManagement.ResourceProviders;
+using UnityEngine.SceneManagement;
+
+namespace Silksong.AssetHelper.CatalogTools
+{
+    internal static class CatalogEntry
+    {
+
+        /// <summary>
+        /// Creates a catalog entry for repacked scene bundles
+        /// </summary>
+        /// <param name="key">The primary key of the bundle.</param>
+        /// <param name="bundlePath">The fully qualified path to the bundle on the filesystem</param>
+        /// <param name="internalBundleName">The name of the AssetBundle asset in the bundle</param>
+        /// <param name="dependencyKeys">List of all the primary keys of the bundle dependencies. Note: All keys found in the dependencies must have their corresponding entry in the catalog</param>
+        public static ContentCatalogDataEntry createBundleEntry(string key, string bundlePath, string internalBundleName, List<string> dependencyKeys)
+        {
+
+            AssetBundleRequestOptions requestOptions = new AssetBundleRequestOptions();
+            requestOptions.AssetLoadMode = AssetLoadMode.RequestedAssetAndDependencies;
+            requestOptions.BundleName = internalBundleName;
+            requestOptions.ChunkedTransfer = false;
+            requestOptions.RetryCount = 0;
+            requestOptions.RedirectLimit = 32;
+            requestOptions.Timeout = 0;
+            requestOptions.BundleSize = 0;
+            requestOptions.ClearOtherCachedVersionsWhenLoaded = false;
+            requestOptions.Crc = 0;
+            requestOptions.UseCrcForCachedBundle = true;
+            requestOptions.BundleSize = (new FileInfo(bundlePath)).Length;
+
+            ContentCatalogDataEntry bundleEntry = new(
+                typeof(IAssetBundleResource),
+                bundlePath,
+                "UnityEngine.ResourceManagement.ResourceProviders.AssetBundleProvider",
+                new object[] { key },
+                dependencyKeys,
+                requestOptions
+
+            );
+
+            return bundleEntry;
+        }
+
+        /// <summary>
+        /// Creates a catalog entry for bundled assets.
```

Comment:
Do you mind expanding on this docstring a bit more? To make it clear that e.g. assetPath is both "the key of the asset within the bundle" and also "the key used to access the asset via ad9s".